test/CppTests/Tests/ExpressionTests.cs

```diff
@@ -460,6 +460,46 @@ public void AssignInvalidExpressionToVariable(ITestSettings settings)
             }
         }
 
+        [Theory]
+        [DependsOnTest(nameof(CompileKitchenSinkForExpressionTests))]
+        [RequiresTestSettings]
+        public void SetExpressionOnVariable(ITestSettings settings)
+        {
+            this.TestPurpose("Call set expression on a variable.");
+            this.WriteSettings(settings);
+
+            IDebuggee debuggee = SinkHelper.Open(this, settings.CompilerSettings, DebuggeeMonikers.KitchenSink.Expression);
+
+            using (IDebuggerRunner runner = CreateDebugAdapterRunner(settings))
+            {
+                this.Comment("Configure launch.");
+                runner.Launch(settings.DebuggerSettings, debuggee, "-fExpression");
+
+                this.Comment("Set a breakpoint so that we can stop at a line.");
+                runner.SetBreakpoints(debuggee.Breakpoints(SinkHelper.Expression, 31));
+
+                this.Comment("Start debugging and hit breakpoint.");
+                runner.Expects.HitBreakpointEvent().AfterConfigurationDone();
+
+                this.Comment("Start setting expressions on variable in frame");
+                using (IThreadInspector threadInspector = runner.GetThreadInspector())
+                {
+                    IFrameInspector currentFrame = threadInspector.Stack.First();
+
+                    string val = runner.SetExpression("myint", "9000", currentFrame.Id);
+                    Assert.Equal("9000", val);
+
+                    string hexVal = runner.SetExpression("myint", "9000", currentFrame.Id, new ValueFormat { hex = true });
+                    Assert.Equal("0x2328", hexVal);
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Verify that fetching the locals returns a `myint` variable with the right value and same with evaluating `myint`?

Reply:
Through you may need Gabrielle's fix for this to work.